src/commands/cmd_query.c

```diff
@@ -247,7 +232,18 @@ static void _ExecuteQuery(void *args) {
 	}
 
 	// in case of an error rollback modifications
-	if(ErrorCtx_EncounteredError()) UndoLog_Rollback(&query_ctx->undo_log);
+	if(ErrorCtx_EncounteredError()) {
+		UndoLog_Rollback(&query_ctx->undo_log);
+		// clear stats so the commad will not be replicated
+		result_set->stats.labels_added           =  0;
+		result_set->stats.nodes_deleted          =  0;
+		result_set->stats.nodes_created          =  0;
+		result_set->stats.properties_set         =  0;
+		result_set->stats.indices_created        =  STAT_NOT_SET;
+		result_set->stats.indices_deleted        =  STAT_NOT_SET;
+		result_set->stats.relationships_created  =  0;
+		result_set->stats.relationships_deleted  =  0;
```

Comment:
Introduce a new function to `resultset` which clears it, probably clearing the records in the set in addition to its statistics, we can't break abstraction and dig into objects.